project-management-infrastructure/src/main/java/life/qbic/projectmanagement/infrastructure/DataManagerVault.java

```diff
@@ -0,0 +1,198 @@
+  private final String envVarKeystoreEntryPassword;
+  private final Path keystorePath;
+
+  public DataManagerVault(@Value("${qbic.security.vault.key.env}") String vaultKeyEnvVar,
+      @Value("${qbic.security.vault.path}") String vaultPathString,
+      @Value("${qbic.security.vault.entry.password.env}") String vaultEntryPassword)
+      throws CertificateException, KeyStoreException, IOException, NoSuchAlgorithmException {
+    if (System.getenv(vaultKeyEnvVar) == null) {
+      throw new DataManagerVaultException(
+          "Cannot find value for environment variable: %s".formatted(vaultKeyEnvVar));
+    }
+    if (System.getenv(vaultEntryPassword) == null) {
+      throw new DataManagerVaultException(
+          "Cannot find value for environment variable: %s".formatted(vaultEntryPassword)
+      );
+    }
+    this.envVarKeystoreEntryPassword = vaultEntryPassword;
+    this.envVarKeystorePassword = vaultKeyEnvVar;
+
+    double entropy;
+    if ((entropy = calculateEntropy(System.getenv(envVarKeystorePassword))) < MIN_ENTROPY) {
+      throw new DataManagerVaultException(
+          "Entry of password for keystore was to low: %f (min: %f)".formatted(entropy,
+              MIN_ENTROPY));
+    }
+    if ((entropy = calculateEntropy(System.getenv(envVarKeystoreEntryPassword))) < MIN_ENTROPY) {
+      throw new DataManagerVaultException(
+          "Entry of password for keystore entries was to low: %f (min: %f)".formatted(entropy,
+              MIN_ENTROPY));
+    }
+
+    this.keystorePath = fromString(vaultPathString);
+    this.keyStore = createVault(vaultKeyEnvVar, keystorePath);
+  }
+
+  // Calculates the product of Shannon entropy and secret length
+  // See https://en.wikipedia.org/wiki/Entropy_(information_theory)
+  private static double calculateEntropy(String secret) {
+    if (secret == null || secret.isEmpty()) {
+      return 0.0;
+    }
+
+    Map<Character, Integer> frequencyMap = new HashMap<>();
+    int length = secret.length();
+
+    // Count character frequencies
+    for (char c : secret.toCharArray()) {
+      frequencyMap.put(c, frequencyMap.getOrDefault(c, 0) + 1);
+    }
+
+    // Compute entropy
+    double entropy = 0.0;
+    for (Integer count : frequencyMap.values()) {
+      double probability = (double) count / length;
+      entropy += probability * (Math.log(probability) / Math.log(2));
+    }
+
+    return -entropy * secret.length(); // Negate since log probabilities are negative
+  }
+
+  private static Path fromString(String path) {
+    Path p = Paths.get(path);
+    if (p.isAbsolute()) {
+      return p;
+    }
+    return Path.of(System.getProperty("user.dir")).resolve(path);
+  }
+
+  private static KeyStore createVault(String vaultKeyEnvVar, Path vaultPath)
+      throws CertificateException, KeyStoreException, IOException, NoSuchAlgorithmException {
+
+    return KeyStore.getInstance(vaultPath.toFile(),
+        System.getenv(vaultKeyEnvVar).toCharArray());
```

Comment:
can you please explain why using System.getenv is relevant here? Why not use envVarKeystorePassword instead?